server/src/main/java/io/crate/auth/HostBasedAuthentication.java

```diff
@@ -172,7 +173,7 @@ Optional<Map.Entry<String, Map<String, String>>> getEntry(@Nullable String user,
             .filter(e -> Matchers.isValidAddress(e.getValue().get(KEY_ADDRESS), connectionProperties.address(), dnsResolver))
             .filter(e -> Matchers.isValidProtocol(e.getValue().get(KEY_PROTOCOL), connectionProperties.protocol()))
             .filter(e -> Matchers.isValidConnection(e.getValue().get(SSL.KEY), connectionProperties))
-            .findFirst();
+            .findFirst(); //TODO!!!!!!!!!!
     }
```

Comment:
Current state is "added boilerplate" and reproduced https://github.com/crate/crate/issues/16059#issuecomment-2137412803 in integraiton tests.

Need to actually implement the fix + add some new (unit) tests